dhcp.nix

```diff
@@ -3,11 +3,81 @@ let
   cfg = config.pi-router;
 in
 {
+  systemd.network.enable = true;
 
-  # TODO Setup DHCP. Maybe KEA?
+  networking = {
+    useDHCP = false;
+    nameservers = cfg.dnsServers;
+    firewall.interfaces."${cfg.interfaces.lan.name}".allowedUDPPorts = [ 67 68 ];
+  };
 
-  # services.dhcpd4 = {
-  #   enable = true;
-  # };
-}
+  # Useful for debugging systemd-networkds and DHCP
+  # journalctl -fu  systemd-networkd.service
+  # systemd.services."systemd-networkd".environment.SYSTEMD_LOG_LEVEL = "debug";
+
+  # See systemd.network(5) for details on the sections below
+  systemd.network.networks.net-lan = {
+
+    # The name of the network interface to match against. ([Match] Block)
+    name = cfg.interfaces.lan.name;
+
+    # [Network] section of the unit.
+    networkConfig = {
+
+      Description = "LAN network";
+
+      # Takes a boolean. If set to "yes", DHCPv4 server will be started.
+      # Defaults to "no". Further settings for the DHCP server may be set in
+      # the [DHCPServer] section described below.
+      DHCPServer = true;
+      # IPMasquerade="ipv4";
+      # Address=10.1.1.1/24
+    };
+
+    # [DHCPServer] section of the unit.
+    dhcpServerConfig = {
+      # EmitDNS = false;
+      # PoolOffset = 50;
+      # EmitDNS = "yes";
+      # ServerAddress = "192.168.101.1/24";
+      DNS = cfg.dnsServers;
```

Comment:
The DHCP server is not advertising our Unbound server as a DNS Server.
We should use `cfg.interfaces.lan.ip` here.

Reply:
```suggestion
      DNS = [ cfg.interfaces.lan.ip ];
```